code/game/machinery/computer/abnormality_ego.dm

```diff
@@ -56,16 +57,39 @@
 				to_chat(usr, span_warning("Not enough PE boxes stored for this operation."))
 				playsound(get_turf(src), 'sound/machines/terminal_prompt_deny.ogg', 50, TRUE)
 				return FALSE
-			var/obj/item/I = new E.item_path(get_turf(src))
+			if(usr.mind.assigned_role == "Extraction Officer")
+				new E.item_path(get_turf(src))
+				to_chat(usr, span_notice("[E.item_path] has been dispensed!"))
+
+			else
+				addtimer(CALLBACK(src, PROC_REF(ShipOut), E.item_path),delay)
+				audible_message(span_notice("[usr.name] has purchased a [E.item_path]"))
+
 			A.stored_boxes -= E.cost
-			A.current_ego += I
```

Comment:
why was this line removed?

Reply:
Ah yes, so basically
This was a temporary feature to only allow X amount of EGO per round.
It was removed because to keep it would increase the shitcode of this PR by a bit, and it was never a feature that existed.

As for if I should remove it there?
I will do it in a future PR as it requires Egor's specific permission, and this PR is an initial step.